subsys/bluetooth/host/hci_core.c

```diff
@@ -4353,6 +4348,11 @@ int bt_disable(void)
 		return err;
 	}
 
+#if defined(CONFIG_BT_RECV_WORKQ_BT)
+	/* Abort RX thread */
+	k_thread_abort(&bt_workq.thread);
```

Comment:
Should we drain the work queue before aborting the thread? Otherwise, moving the abort may have no effect, depending on system priorities.

Reply:
We could also leave the thread instead of aborting it. An idle thread should not be a problem, right?

Reply:
We call `k_work_queue_init();` and `k_work_queue_start()` on `bt_enable()`, so it is probably better to cleanup all state.
I agree with you that we should consider doing a proper cleanup. But that I would leave out of the scope of this PR.

Reply:
> We call `k_work_queue_init();` and `k_work_queue_start()` on `bt_enable()`, so it is probably better to cleanup all state.
> 
> I agree with you that we should consider doing a proper cleanup. But that I would leave out of the scope of this PR.

I agree. The current version of this PR is simply reverting the functionality to what it used to be before the HCI API PR.